packages/echo-core/src/services/analytics/analyticsModule.ts

```diff
@@ -43,8 +43,15 @@ export function analyticsSetUserCompany(company: string): void {
 export class AnalyticsModule {
     moduleName: string;
     offlineTracker: OfflineTracker;
-    constructor(moduleName: string) {
+    staticEventProperties?: AnalyticsPropertyTypes;
+    staticErrorProperties?: AnalyticsPropertyTypes;
+    constructor(
+        moduleName: string,
+        args?: { staticEventProperties?: AnalyticsPropertyTypes; staticErrorProperties?: AnalyticsPropertyTypes }
+    ) {
         this.moduleName = moduleName;
+        this.staticEventProperties = args?.staticEventProperties;
```

Comment:
Just to better understand: what purpose do these static props serve? Just to be able to add more information to the error object?

Reply:
echo3d wants to add versionNumber to all of their events, and this one is static throughout the application lifetime. So instead of attaching it for all events and errors, they add it when creating the analytics module